packages/launchpad/src/signingcosmosclient.ts

```diff
@@ -88,8 +88,8 @@ export class SigningCosmosClient extends CosmosClient {
   public async signAndBroadcast(msgs: readonly Msg[], fee: StdFee, memo = ""): Promise<BroadcastTxResult> {
     const { accountNumber, sequence } = await this.getSequence();
     const chainId = await this.getChainId();
-    const signBytes = makeSignBytes(msgs, fee, chainId, memo, accountNumber, sequence);
-    const signature = await this.signer.sign(this.senderAddress, signBytes);
+    const signDoc = makeSignDoc(msgs, fee, chainId, memo, accountNumber, sequence);
```

Comment:
If I understood this correctly above, the `sign` method can now modify the SignDoc and returns the updated one. In particular, it could adjust fee and memo to be different than what was passed in.

Right below this, when building `signedTx`, we should use the returned (signed) fee and memo. 

This should fail if we test it with an OfflineSigner implementation that eg. doubles the fees and sets memo to "Hello World" before signing. It would be good to test that case and ensure that all code handles updated signdocs results properly.

Reply:
I am fine with only passing the original messages in here, as if the wallet changed those when signing, we probably want this to fail. But `fee` and `memo` should definitely be under the control of the OfflineSigner